lib/matplotlib/axis.py

```diff
@@ -99,38 +99,28 @@ def __init__(self, axes, loc, label,
 
         self._loc = loc
 
+        major_minor = "major" if major else "minor"
+
         if size is None:
-            if major:
-                size = rcParams['%s.major.size' % name]
-            else:
-                size = rcParams['%s.minor.size' % name]
+            size = rcParams[f"{name}.{major_minor}.size"]
         self._size = size
 
         if width is None:
-            if major:
-                width = rcParams['%s.major.width' % name]
-            else:
-                width = rcParams['%s.minor.width' % name]
+            width = rcParams[f"{name}.{major_minor}.width"]
         self._width = width
 
         if color is None:
-            color = rcParams['%s.color' % name]
-        self._color = color
+            color = rcParams[f"{name}.color"]
 
         if pad is None:
-            if major:
-                pad = rcParams['%s.major.pad' % name]
-            else:
-                pad = rcParams['%s.minor.pad' % name]
+            pad = rcParams[f"{name}.major.pad"]
```

Comment:
Doesn't this also need the `major_minor` field?

Reply:
oops, fixed